src/dotty/tools/backend/jvm/CollectSuperCalls.scala

```diff
@@ -17,14 +17,14 @@ class CollectSuperCalls extends MiniPhaseTransform {
 
   def phaseName: String = "collectSuperCalls"
 
-  override def transformSuper(tree: Super)(implicit ctx: Context, info: TransformerInfo): Tree = {
+  override def transformSelect(tree: tpd.Select)(implicit ctx: Context, info: TransformerInfo): tpd.Tree = {
     tree match {
-      case Trees.Super(qual: This, mix) if mix.nonEmpty =>
+      case Select(Super(qual, mix), _) if mix.nonEmpty =>
         val classSymbol = qual.symbol.asClass.classSymbol
-        registerSuperCall(classSymbol, tree.tpe.baseClasses.head)
+        registerSuperCall(classSymbol, tree.symbol.owner.asClass)
       case _ =>
     }
-    super.transformSuper(tree)
+    super.transformSelect(tree)
```

Comment:
Just return `tree`, the default implementation of `transformX` is always the identity